### DISCUSSION TITLE
Field picklist shown on front-end

### DISCUSSION THREAD

**Discussion by MicheleAmoroso** [OP-TED/eForms-SDK#806] [Q&A] [answered]
Hi, is it possible to obtain information from the SDK regarding the picklist of fields shown on the frontend on eNotices2?
For example, OPT-300-Procedure-Buyer field shows values ​​taken from BT-500-Organization-Company. Is this information possible to obtain somewhere for other similar fields?

**Comment by bertrand-lorentz** [comment, marked as answer]
I'll try to go through the steps for OPT-300-Procedure-Buyer, based on the content of `fields.json`.
Those steps can be followed for other fields that play a similar role: referencing another "thing" (organisation, lot, etc.) in the notice.

The fact that OPT-300-Procedure-Buyer contains an reference to an identifier is indicated by its type "id-ref".
The sets of identifiers it can reference is indicated in the `idSchemes` property. In this case it's only "ORG".

The field that defines the "ORG" identifiers for the notice is the one that has the corresponding value in the `idScheme` (not plural). In this case it is OPT-200-Organization-Company.
So the list of possible values for OPT-300-Procedure-Buyer is the various ORG-xxxx that are in the various occurences of OPT-200-Organization-Company.

Displaying a list of ORG-0001, ORG-0002, etc. is not helpful for the user, so we have indicated which other field can be used as a helpful short text, or "caption", to help the user identify the organisations.
It can be found by looking for the node which has OPT-200-Organization-Company as its identifier field (property `identifierFieldId`). In this case it is ND-Organization, and this node has the property `captionFieldId` with the value BT-500-Organization-Company, which corresponds to the organisation name.
So for each organisation in the notice, you can have a list of identifiers and their corresponding captions.

So the list displayed to the user can show the values from the caption, and when one is selected, the corresponding identifiers can be put in the XML notice.

See this pages in the docs for more details on the content of fields.json.
